File: app/src/main/java/ru/iandreyshev/parserrss/models/repository/Rss.java
package ru.iandreyshev.parserrss.models.repository;

import android.support.annotation.NonNull;

import java.util.ArrayList;
import java.util.List;

import io.objectbox.annotation.Entity;
import io.objectbox.annotation.Id;
import io.objectbox.annotation.Index;
import io.objectbox.annotation.Transient;

@Entity
public final class Rss {
    @Id
    long mId;

    String mTitle = "";
    @Index
    String mUrl = "";
    String mOrigin = "";
    String mDescription;

    @Transient
    private List<Article> mArticles = new ArrayList<>();

    public Rss(@NonNull final String title, @NonNull final String origin) {
        mTitle = title;
        mOrigin = origin;
    }

    Rss() {
    }

    public long getId() {
        return mId;
    }

    @NonNull
    public String getTitle() {
        return mTitle;
    }

    public String getDescription() {
        return mDescription;
    }

    @NonNull
    public String getUrl() {
        return mUrl;
    }

    @NonNull
    public String getOrigin() {
        return mOrigin;
    }

    @NonNull
    public List<Article> getArticles() {
        return mArticles;
    }

    public void setId(long newId) {
        mId = newId;
    }

    public void setUrl(final String url) {
        mUrl = url;
    }

    public void setArticles(@NonNull final List<Article> newArticles) {
        mArticles = new ArrayList<>(newArticles);
    }

    public void setDescription(final String description) {
        mDescription = description;
    }

    @Override
    public final boolean equals(Object other) {
        return (other instanceof Rss) && mUrl.equals(((Rss) other).mUrl);
    }

    @Override
    public final int hashCode() {
        return mUrl.hashCode();
    }
}

File: app/src/main/java/ru/iandreyshev/parserrss/models/repository/Rss.kt
package ru.iandreyshev.parserrss.models.repository

import java.util.ArrayList

import io.objectbox.annotation.Entity
import io.objectbox.annotation.Id
import io.objectbox.annotation.Index
import io.objectbox.annotation.Transient

@Entity
data class Rss (
        @Id var id: Long = 0,
        var title: String = "",
        @Index var url: String = "",
        var origin: String = "",
        var description: String? = null) {

    @Transient
    var articles: List<Article> = ArrayList()
        set(newArticles) {
            field = ArrayList(newArticles)
        }

    override fun equals(other: Any?): Boolean {
        return other is Rss && url == other.url
    }

    override fun hashCode(): Int {
        return url.hashCode()
    }
}
